pkg/docker/config/config.go

```diff
@@ -502,26 +505,26 @@ func listAuthsFromCredHelper(credHelper string) (map[string]string, error) {
 
 // getPathToAuth gets the path of the auth.json file used for reading and writing credentials
 // returns the path, and a bool specifies whether the file is in legacy format
-func getPathToAuth(sys *types.SystemContext) (string, bool, error) {
+func getPathToAuth(sys *types.SystemContext) (authPath, error) {
 	return getPathToAuthWithOS(sys, runtime.GOOS)
 }
 
 // getPathToAuthWithOS is an internal implementation detail of getPathToAuth,
 // it exists only to allow testing it with an artificial runtime.GOOS.
-func getPathToAuthWithOS(sys *types.SystemContext, goOS string) (string, bool, error) {
+func getPathToAuthWithOS(sys *types.SystemContext, goOS string) (authPath, error) {
 	if sys != nil {
 		if sys.AuthFilePath != "" {
-			return sys.AuthFilePath, false, nil
+			return newAuthPathDefault(sys.AuthFilePath), nil
 		}
 		if sys.LegacyFormatAuthFilePath != "" {
-			return sys.LegacyFormatAuthFilePath, true, nil
+			return authPath{path: sys.LegacyFormatAuthFilePath, legacyFormat: true}, nil
```

Comment:
This is the only place that actually set `legacyFormat: true` that I saw, but please double check.  Everything else should be using `newAuthPathDefault`.

Reply:
There’s also `dockerLegacyHomePath`.

Reply:
OK right, which is also correctly not changed.